xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp

```diff
@@ -500,27 +501,41 @@ bool CDVDVideoCodecAndroidMediaCodec::Open(CDVDStreamInfo &hints, CDVDCodecOptio
       }
       break;
     case AV_CODEC_ID_HEVC:
+    {
       if (m_hints.profile == FF_PROFILE_HEVC_REXT)
       {
         // No known h/w decoder supporting Hi10P
         goto FAIL;
       }
 
-      if (m_hints.codec_tag == MKTAG('d', 'v', 'h', 'e'))
-      {
-        m_mime = "video/dolby-vision";
-        m_formatname = "amc-dvhe";
-      }
-      else if (m_hints.codec_tag == MKTAG('d', 'v', 'h', '1'))
-      {
-        m_mime = "video/dolby-vision";
-        m_formatname = "amc-dvh1";
-      }
-      else
+      m_mime = "video/hevc";
+      m_formatname = "amc-hevc";
+
+      bool isDvhe = (m_hints.codec_tag == MKTAG('d', 'v', 'h', 'e'));
+      bool isDvh1 = (m_hints.codec_tag == MKTAG('d', 'v', 'h', '1'));
+
+      if (isDvhe || isDvh1)
       {
-        m_mime = "video/hevc";
-        m_formatname = "amc-hevc";
+        bool displaySupportsDovi = CAndroidUtils::GetDisplayHDRCapabilities().SupportsDolbyVision();
+        bool mediaCodecSupportsDovi =
+            CAndroidUtils::SupportsMediaCodecMimeType("video/dolby-vision");
+
+        CLog::Log(LOGDEBUG,
+                  "CDVDVideoCodecAndroidMediaCodec::Open Dolby Vision support: "
+                  "Display {}, MediaCodec: {}",
+                  displaySupportsDovi, mediaCodecSupportsDovi);
+
+        if (displaySupportsDovi && mediaCodecSupportsDovi)
+        {
+          m_mime = "video/dolby-vision";
+
+          if (isDvhe)
```

Comment:
Good place for ternary operator 👌

`m_formatname = isDvhe ? "amc-dvhe" : "amc-dvh1";`